equinix/resource_metal_reserved_ip_block.go

```diff
@@ -215,12 +216,12 @@ func resourceMetalReservedIPBlock() *schema.Resource {
 	}
 	// TODO: add comments field, used for reservations that are not automatically approved
 	return &schema.Resource{
-		Create: resourceMetalReservedIPBlockCreate,
-		Read:   resourceMetalReservedIPBlockRead,
-		Update: resourceMetalReservedIPBlockUpdate,
-		Delete: resourceMetalReservedIPBlockDelete,
+		CreateContext:        diagnosticsWrapper(resourceMetalReservedIPBlockCreate),
+		ReadWithoutTimeout:   diagnosticsWrapper(resourceMetalReservedIPBlockRead),
+		UpdateWithoutTimeout: diagnosticsWrapper(resourceMetalReservedIPBlockUpdate),
```

Comment:
I think it would be good to have a timeout on update and delete; is there a platform limitation or other reason why we can't/shouldn't support timeouts on those operations?

Reply:
Since the default timeout has only been configured/exposed for `create` the create with context is used
`Timeouts: &schema.ResourceTimeout{
    Create: schema.DefaultTimeout(ReservedIPCreateTimeout),
},`